libs/core/src/components/spinner/gds-spinner.ts

```diff
@@ -0,0 +1,173 @@
+import { property } from 'lit/decorators.js'
+
+import { GdsElement } from '../../gds-element'
+import { tokens } from '../../tokens.style'
+import { TransitionalStyles } from '../../transitional-styles'
+import {
+  gdsCustomElement,
+  html,
+} from '../../utils/helpers/custom-element-scoping'
+import { styles } from './gds-spinner.styles'
+
+/**
+ * GdsSpinner Component
+ * A loading indicator web component that supports various sizes and display modes.
+ *
+ * @extends {GdsElement}
+ *
+ * @example
+ * ```html
+ * <!-- Basic usage -->
+ * <gds-spinner size="md" text="Loading..."></gds-spinner>
+ *
+ * <!-- Screen reader only text -->
+ * <gds-spinner size="md" text="Loading..." .visualText=${false}></gds-spinner>
+ *
+ * <!-- Cover container -->
+ * <gds-spinner size="md" text="Loading..." cover></gds-spinner>
+ * ```
+ *
+ * @csspart spinner - The spinning circle element
+ * @csspart text - The loading text element
+ *
+ * @fires gds-element-disconnected - Fired when the element is removed from the DOM
+ */
+@gdsCustomElement('gds-spinner')
+export class GdsSpinner extends GdsElement {
+  /**
+   * Component styles combining design tokens, host styles, and component-specific styles
+   * @type {Array<CSSStyleSheet>}
+   */
+  static styles = [
+    tokens,
+    css`
+      :host {
+        display: block;
+      }
+    `,
+    styles,
+  ]
+
+  /**
+   * Loading text to be displayed. Can be shown visually or for screen readers only.
+   * @type {string|undefined}
+   */
+  @property({ type: String })
+  text?: string
+
+  /**
+   * Controls text visibility mode.
+   * When true, text is visible and announced to screen readers.
+   * When false, text is only announced to screen readers.
+   * @type {boolean}
+   * @default true
+   */
+  @property({ type: Boolean })
+  visualText: boolean = true
+
+  /**
+   * When true, adds a semi-transparent backdrop over the container.
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  cover: boolean = false
+
+  /**
+   * When true, spinner covers the entire viewport with backdrop.
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  fullscreen: boolean = false
+
+  /**
+   * Controls the size of the spinner.
+   * @type {'default'|'sm'|'md'|'lg'}
+   * @default 'default'
+   */
+  @property({ type: String })
+  size: 'default' | 'md' | 'lg' | 'sm' = 'default'
+
+  /**
+   * Lifecycle method called when component is added to DOM.
+   * Sets up accessibility attributes and applies transition styles.
+   * @return {void}
+   */
+  connectedCallback(): void {
+    super.connectedCallback()
+    this.setAttribute('role', 'status')
+    this.setAttribute('aria-busy', 'true')
+    TransitionalStyles.instance.apply(this, 'gds-spinner')
+  }
+
+  /**
+   * Generates CSS classes for the wrapper element based on component state.
+   * @private
+   * @return {string} Space-separated class names
+   */
+  private _getWrapperClasses(): string {
```

Comment:
Here, I would return a object that maps the booleans, which can be fed to Lit's `[classMap](https://lit.dev/docs/templates/directives/#classmap)` directive.

Also, by convention, we use native JS private field syntax were possible, ie `#field` rather than `private _field`. [Here's the rationale.](http://localhost:4400/?path=/docs/contributing-coding-guidelines--docs#-for-private-properties)

I.e, something like
```ts
#getWrapperClasses = () => ({
  'gds-spinner-fullscreen': this.fullscreen,
 ...etc
})
```